core/base/src/TObject.cxx

```diff
@@ -778,6 +779,36 @@ void TObject::SavePrimitiveConstructor(std::ostream &out, TClass *cl, const char
    out << variable_name << " = new " << cl->GetName() << "(" << constructor_agrs << ");" << std::endl;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Save array in the output stream "out".
+/// Create unique variable name based on prefix value
+
+TString TObject::SavePrimitiveArray(std::ostream &out, const char *prefix, Int_t len, Double_t *arr)
+{
+   static int arrid = 0;
```

Comment:
```suggestion
   thread_local int arrid = 0;
```
to work toward `SavePrimitive` begin callable from 2 distinct threads on 2 distinct objects.